data-prepper-api/src/main/java/org/opensearch/dataprepper/model/sink/AbstractSink.java

```diff
@@ -34,6 +35,11 @@ public AbstractSink(final PluginSetting pluginSetting, int numRetries, int waitT
         retryThread = null;
         this.maxRetries = numRetries;
         this.waitTimeMs = waitTimeMs;
+        sinkContext = (SinkContext)pluginSetting.getAttributeOrDefault("sink_context", null);
```

Comment:
We are trying to move away from using the `PluginSetting` class. So let's not rely on it further.

I think you don't need any changes to `AbstractSink` here.